src/chunk.c

```diff
@@ -2151,23 +2266,143 @@ ts_chunk_drop(Chunk *chunk, DropBehavior behavior, int32 log_level)
-	ts_chunk_delete_by_relid(chunk->table_id, behavior);
+	ts_chunk_delete_by_relid(chunk->table_id, behavior, preserve_catalog_row);
 
 	/* Drop the table */
 	performDeletion(&objaddr, behavior, 0);
 }
 
+static void
+ts_chunk_drop_process_materialization(Oid hypertable_relid,
+									  CascadeToMaterializationOption cascade_to_materializations,
+									  Datum older_than_datum, Oid older_than_type,
+									  Oid newer_than_type, Chunk *chunks, int num_chunks)
+{
+	Dimension *time_dimension;
+	int64 older_than_time;
+	int64 ignore_invalidation_older_than;
+	int64 minimum_invalidation_time;
+	int64 lowest_completion_time;
+	List *continuous_aggs;
+	ListCell *lc;
+	Cache *hcache;
+	Hypertable *ht;
+	int i;
+
+	FormData_continuous_agg cagg;
+
+	/* nothing to do if also dropping materializations */
+	if (cascade_to_materializations == CASCADE_TO_MATERIALIZATION_TRUE)
+		return;
+
+	Assert(cascade_to_materializations == CASCADE_TO_MATERIALIZATION_FALSE);
+
+	if (OidIsValid(newer_than_type))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("cannot use newer_than parameter to drop_chunks with "
+						"cascade_to_materializations")));
+
+	if (!OidIsValid(older_than_type))
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("must use older_than parameter to drop_chunks with "
+						"cascade_to_materializations")));
+
+	hcache = ts_hypertable_cache_pin();
+	ht = ts_hypertable_cache_get_entry(hcache, hypertable_relid);
+	if (!ht)
+		elog(ERROR, "can only call drop_chunks on hypertables");
+
+	time_dimension = hyperspace_get_open_dimension(ht->space, 0);
+	older_than_time = get_internal_time_from_endpoint_specifiers(ht->main_table_relid,
+																 time_dimension,
+																 older_than_datum,
+																 older_than_type,
+																 "older_than",
+																 "drop_chunks");
+	ignore_invalidation_older_than =
+		ts_continuous_aggs_max_ignore_invalidation_older_than(ht->fd.id, &cagg);
+	minimum_invalidation_time =
+		ts_continuous_aggs_get_minimum_invalidation_time(ts_get_now_internal(time_dimension),
+														 ignore_invalidation_older_than);
+
+	/* minimum_invalidation_time is inclusive; older_than_time is exclusive */
+	if (minimum_invalidation_time < older_than_time)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("older_than must be greater than the ignore_invalidation_older_than "
+						"parameter of %s.%s",
+						cagg.user_view_schema.data,
+						cagg.user_view_name.data)));
+
+	/* error for now, maybe better as a warning and ignoring the chunks? */
+	/* We cannot move a completion threshold up transactionally without taking locks
+	 * that would block the system. So, just bail. The completion threshold
+	 * should be much higher than this anyway */
+	lowest_completion_time = ts_continuous_aggs_min_completed_threshold(ht->fd.id, &cagg);
+	if (lowest_completion_time < older_than_time)
+		ereport(ERROR,
```

Comment:
do we have a test case for this?

Reply:
yes in tsl/test/expected/continuous_aggs_ddl-11.out (around line 622)